File: app/frontend/src/app/game/components/2_navigasi_menu/2_navigasi_bawah/5_geopolitik/2_organisasi_internasional/1_organisasi_PBB/logic/python/membership_movement/1_join_logic/JoinHandler.ts
/**
 * JoinHandler TS Bridge: Matches the logic in join_handler.py
 */
export class JoinHandler {
    private probability: number;

    constructor(baseProbability = 0.40) {
        this.probability = baseProbability;
    }

    public evaluateJoin(countryName: string, decisionWeight: number): boolean {
        if (decisionWeight > 60) {
            const bonus = (decisionWeight - 60) / 100;
            const adjustedProb = Math.min(this.probability + bonus, 0.80);
            
            return Math.random() < adjustedProb;
        }
        return false;
    }
}

File: app/frontend/src/app/game/components/2_navigasi_menu/2_navigasi_bawah/5_geopolitik/2_organisasi_internasional/1_organisasi_PBB/logic/python/membership_movement/1_join_logic/join_handler.py
import random

class JoinHandler:
    """
    Handles the specific logic for an AI country to join an organization.
    """
    def __init__(self, base_probability=0.40):
        self.probability = base_probability

    def evaluate_join(self, country_name, decision_weight):
        """
        Returns True if the country decides to join.
        """
        if decision_weight > 60:
            # High weight increases probability slightly
            bonus = (decision_weight - 60) / 100
            adjusted_prob = min(self.probability + bonus, 0.80)
            
            if random.random() < adjusted_prob:
                return True
        return False
